File: reddio-csharp/reddio.api/V1/Rest/ResponseWrapper.cs
using Newtonsoft.Json;

namespace Reddio.Api.V1.Rest
{
    public class ResponseWrapper<T>
    {
        [JsonProperty("status")] public string Status { get; set; }
        [JsonProperty("error")] public string Error { get; set; }
        [JsonProperty("data")] public T Data { get; set; }

        public ResponseWrapper()
        {
        }

        public ResponseWrapper(string status, string error, T data)
        {
            Status = status;
            Error = error;
            Data = data;
        }
    }
}

File: reddio-java/reddio-api/src/main/java/com/reddio/api/v1/rest/ResponseWrapper.java
package com.reddio.api.v1.rest;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor(staticName = "of")
public class ResponseWrapper<T> {
    @JsonProperty("status")
    private String status;

    @JsonProperty("error")
    private String error;

    @JsonProperty("error_code")
    private Integer errorCode;

    @JsonProperty("data")
    private T data;

}
